### PR TITLE
SONARGO-52 Create a new repository for common Gradle scripts

### PR DESCRIPTION
**Pull request by petertrr**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+Thumbs.db
+# Folder config file
+Desktop.ini
+
+# ---- IntelliJ IDEA
+*.iws
+*.iml
+*.ipr
+.idea/
+.run/
```
Comment (mstachniuk) [inline .gitignore:30]: please add a new line at the end

File: LICENSE [hunk 2]
```diff
@@ -1,165 +1,184 @@
...
-received it does not specify a version number of the GNU Lesser
-General Public License, you may choose any version of the GNU Lesser
-General Public License ever published by the Free Software Foundation.
-
-  If the Library as you received it specifies that a proxy can decide
-whether future versions of the GNU Lesser General Public License shall
-apply, that proxy's public statement of acceptance of any version is
-permanent authorization for you to choose that version for the
-Library.
+SONAR Source-Available License v1.0
```
Comment (mstachniuk) [inline LICENSE:1]: Why this license and not LGPL? It is not an analyzer? Maybe it should be discussed with someone?

File: build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+ * modify it under the terms of the Sonar Source-Available License Version 1, as published by SonarSource SA.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the Sonar Source-Available License for more details.
+ *
+ * You should have received a copy of the Sonar Source-Available License
+ * along with this program; if not, see https://sonarsource.com/license/ssal/
+ */
```
Comment (mstachniuk) [inline build.gradle.kts:16]: It is expected to have this file empty?

File: build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+/*
+ * SonarQube Text Plugin
+ * Copyright (C) 2021-2024 SonarSource SA
```
Comment (mstachniuk) [inline build.gradle.kts:3]: I guess the year should be 2024 as the repos was created in 2024 ;-)

File: gradle-modules/src/main/kotlin/org.sonarsource.cloud-native.java-conventions.gradle.kts [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+tasks.withType<JavaCompile> {
+    options.encoding = "UTF-8"
+}
+
+tasks.withType<Javadoc> {
+    options.encoding = "UTF-8"
+    options {
+        (this as CoreJavadocOptions).addStringOption("Xdoclint:none", "-quiet")
+    }
+}
```
Comment (mstachniuk) [inline gradle-modules/src/main/kotlin/org.sonarsource.cloud-native.java-conventions.gradle.kts:37]: What about jacoco configuration? In IaC and PHP it I configured in this file

**Review by mstachniuk — APPROVED** [review]
LGTM and please see my comments

File: gradle-modules/src/main/kotlin/org.sonarsource.cloud-native.java-conventions.gradle.kts (hunk 5 above)
Comment (petertrr) [reply gradle-modules/src/main/kotlin/org.sonarsource.cloud-native.java-conventions.gradle.kts:37]: Good catch

File: build.gradle.kts (hunk 3 above)
Comment (petertrr) [reply build.gradle.kts]: Removed

File: LICENSE (hunk 2 above)
Comment (petertrr) [reply LICENSE]: Changed back to LGPL and also updated project name

**Pull request merged**